src/server/middleware/missingBundleMiddleware.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ */
+const MultiCompiler = require('webpack/lib/MultiCompiler');
+const dedent = require('dedent');
+
+function missingBundleMiddleware(compiler) {
+  return (req, res, next) => {
+    // In multi compiler mode, all entry points should be available
+    if (compiler instanceof MultiCompiler) {
+      next();
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    const [_, platform] = req.path.match(/(?:\.(ios|android))?\.bundle/) || [];
+
+    if (platform) {
+      res.status(500).end(
+        dedent`
+        Bundle couldn't be loaded.
+
+        Try the following to fix the issue:
+        - Ensure that Haul is running with --platform all or --platform ${platform}
```

Comment:
Yeah, but wanted to keep it simple for now. I think that error message as is should be enough for typical user to understand where is the problem.